### PR TITLE
Migrate to SLF4J for logging

### LINKED ISSUE #1116: Use SLF4J throughout plugin
java.util.Logger is used in many places, but maven builds on SLF4J. The output from the plugin is inconsistent (for example, pom sorting).

```
[INFO] --- spotless-maven-plugin:2.20.1:check (default) @ parent-pom ---
[INFO] Up-to-date checking enabled
[INFO] Index file does not exist. Fallback to an empty index
Feb 03, 2022 1:24:36 PM com.diffplug.spotless.glue.pom.SortPomFormatterFunc$MySortPomLogger info
INFO: Sorting file ...\pom2973277670321689996.xml
Feb 03, 2022 1:24:37 PM com.diffplug.spotless.glue.pom.SortPomFormatterFunc$MySortPomLogger info
INFO: Pom file is already sorted, exiting
[INFO] 
```

All places in the code could be updated to use SLF4J's Logger instead.

Gradle looks like that framework builds on SLF4J too: https://docs.gradle.org/current/javadoc/org/gradle/api/logging/Logger.html

### PR DESCRIPTION
**Pull request by jamietanna**
Closes #1116.

### CODE REVIEW

File: lib/src/main/java/com/diffplug/spotless/FormatExceptionPolicyLegacy.java
Comment (jamietanna) [inline lib/src/main/java/com/diffplug/spotless/FormatExceptionPolicyLegacy.java:24]: :question: IntelliJ/SonarLint warns that we're using the wrong class here - should we amend this to:
```suggestion
	private static final Logger logger = LoggerFactory.getLogger(FormatExceptionPolicyLegacy.class.getName());
```

**Review by nedtwigg — APPROVED** [review]
Thanks very much, this LGTM. I'd like one other committer to sign-off on this before I press merge though.

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
Comment (lutovich) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java:130]: Perhaps use a `{}` placeholder here instead of `mavenCoords`

**Review by lutovich — APPROVED** [review]
Looks good to me 👍

Just curious, why do we use `LoggerFactory.getLogger(Type.class.getName())` instead of `LoggerFactory.getLogger(Type.class)`?

**Comment by nedtwigg** [comment]
@jamietanna can you address the points @lutovich rasied above?

**Comment by jamietanna** [comment]
> Just curious, why do we use `LoggerFactory.getLogger(Type.class.getName())` instead of `LoggerFactory.getLogger(Type.class)`?

I'd kept this in line with the existing code to do this, as I'd assumed that there was a reason behind it. Looking at the code for `java.util.logging.Logger#getLogger`, it only takes a `String`, whereas we can provide a `Class<?>` with SLF4J.

I'm happy amending it, as it then looks more common for SLF4J, and that can then perform the `.getName()` call itself?

**Comment by nedtwigg** [comment]
If we're adopting log4j, let's do it canonically. Also take a look at "Perhaps use a {} placeholder here instead of mavenCoords". Thanks!

**Comment by jamietanna** [comment]
Done, thanks for the speedy reply!

**Comment by nedtwigg** [comment]
Thanks! In the future, please add new commits instead of force-pushing, easier to review :)

**Comment by jamietanna** [comment]
Sorry about that. I would've `fixup`'d if I'd have known. Will remember in the future :smile:

**Comment by lutovich** [comment]
Looks good. Thanks for addressing my comments!

**Comment by nedtwigg** [comment]
Thanks a ton for taking on the SLF4J migration @jamietanna!

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.3.0`, not enough new features to justify a maven release yet.

**Pull request merged**
